traveler-swift-core/TravelerKit/Models/CancellationRequest.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  CancellationRequest.swift
+//  TravelerKit
+//
+//  Created by Ben Ruan on 2019-10-03.
+//  Copyright © 2019 Ata Namvari. All rights reserved.
+//
+
+import Foundation
+
+public struct CancellationRequest {
+    public let quote: CancellationQuote
+    public let reason: CancellationReason
+    public let explanation: String?
+
+    public init(quote: CancellationQuote, reason: CancellationReason, explanation: String?) throws {
+        guard quote.cancellationReasons.contains(where: { $0.id == reason.id }) else {
+            throw CancellationError.explanationRequired
```

Comment:
You need a different error here!

This doesnt have anything to do with explanation does it?